ddtrace/settings/config.py

```diff
@@ -471,11 +472,14 @@ def __init__(self):
 
         self.env = _get_config("DD_ENV", self.tags.get("env"))
         self.service = _get_config("DD_SERVICE", self.tags.get("service", DEFAULT_SPAN_SERVICE_NAME))
+        self._inferred_service = detect_service(sys.argv)
```

Comment:
"Inferred service" is a name collision that will add some confusion, we should probably name it something else.

It collides with "inferred service" as part of peer service when the library is talking (receiving or sending) to a third-party system.

Maybe `auto_named_service` or `_inferred_base_service` or something similar?